src/main/java/com/configcat/ConfigFetcher.java

```diff
@@ -99,55 +95,60 @@ private CompletableFuture<FetchResponse> executeFetchAsync(int executionCount) {
         });
     }
 
-    private CompletableFuture<FetchResponse> getResponseAsync() {
-        if (this.currentFuture != null && !this.currentFuture.isDone()) {
-            this.logger.debug("Config fetching is skipped because there is an ongoing fetch request");
-            return this.currentFuture;
-        }
-
-        Config cachedConfig = this.configJsonCache.readFromCache();
-        Request request = this.getRequest(cachedConfig.eTag);
+    private CompletableFuture<FetchResponse> getResponseAsync(final String eTag) {
+        Request request = this.getRequest(eTag);
         CompletableFuture<FetchResponse> future = new CompletableFuture<>();
         this.httpClient.newCall(request).enqueue(new Callback() {
             @Override
             public void onFailure(@NotNull Call call, @NotNull IOException e) {
+                String message = "An error occurred during fetching of the latest config.json.";
```

Comment:
Maybe it should be the same as above. `Exception while trying to fetch the config.json.`